packages/warehouses/src/warehouseClients/DatabricksWarehouseClient.ts

```diff
@@ -189,54 +214,50 @@ export class DatabricksWarehouseClient implements WarehouseClient {
             table: string;
         }[],
     ) {
-        let pool: odbc.Pool;
-        let results: Result<SparkSchemaResult>[];
-        try {
-            pool = await odbc.pool(this.connectionString);
-        } catch (e) {
-            throw new WarehouseConnectionError(e.message);
-        }
+        const { session, close } = await this.getSession();
+        let results: SchemaResult[][];
+
         try {
             const promises = requests.map(async (request) => {
-                let connection: odbc.Connection;
+                let query: IOperation | null = null;
                 try {
-                    connection = await pool.connect();
-                } catch (e) {
-                    throw new WarehouseConnectionError(e.message);
-                }
-                try {
-                    const columns = (await connection.columns(
-                        // @ts-ignore
-                        'SPARK', // This is always SPARK
-                        request.schema,
-                        request.table,
-                        '',
-                    )) as Result<SparkSchemaResult>;
-                    return columns;
+                    query = await session.getColumns({
+                        catalogName: request.database,
+                        schemaName: request.schema,
+                        tableName: request.table,
+                    });
+
+                    const result = (await query.fetchAll()) as SchemaResult[];
+
+                    return result;
                 } catch (e) {
                     throw new WarehouseQueryError(e.message);
                 } finally {
-                    await connection.close();
+                    if (query) query.close();
                 }
             });
             results = await Promise.all(promises);
         } finally {
-            await pool.close();
+            await close();
         }
+
+        const catalog = this.catalog || 'DEFAULT';
         return results.reduce<WarehouseCatalog>(
             (acc, result, index) => {
                 const columns = Object.fromEntries<DimensionType>(
                     result.map((col) => [
                         col.COLUMN_NAME,
-                        mapFieldType(col.TYPE_NAME),
+                        mapFieldType(col.DATA_TYPE),
```

Comment:
I think something is going wrong in here as boolean/dates are being marked as strings

![Screenshot 2022-10-18 at 11 19 34](https://user-images.githubusercontent.com/9117144/196404301-15dcdf7f-89f8-477a-b814-2d874bcbacc4.png)

Reply:
@ZeRego good catch, hmm, that mapping does not look reliable. I've reverted to existing one and added missing fields.

<img width="406" alt="CleanShot 2022-10-18 at 14 54 22@2x" src="https://user-images.githubusercontent.com/962095/196411897-d01010bc-f97b-4891-b2ef-763f0682e218.png">

```
{ type: 'BIGINT', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'STRING', mapped: 'string' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'STRING', mapped: 'string' }
{ type: 'STRING', mapped: 'string' }
{ type: 'TIMESTAMP', mapped: 'timestamp' }
{ type: 'DATE', mapped: 'date' }
{ type: 'DATE', mapped: 'date' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'INTERVAL DAY', mapped: 'string' }
{ type: 'INT', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'DATE', mapped: 'date' }
{ type: 'STRING', mapped: 'string' }
{ type: 'BOOLEAN', mapped: 'boolean' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'TIMESTAMP', mapped: 'timestamp' }
{ type: 'TIMESTAMP', mapped: 'timestamp' }
{ type: 'TIMESTAMP', mapped: 'timestamp' }
{ type: 'DATE', mapped: 'date' }
{ type: 'STRING', mapped: 'string' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'STRING', mapped: 'string' }
{ type: 'STRING', mapped: 'string' }
{ type: 'TIMESTAMP', mapped: 'timestamp' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'STRING', mapped: 'string' }
{ type: 'DOUBLE', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'BIGINT', mapped: 'number' }
{ type: 'DATE', mapped: 'date' }
{ type: 'STRING', mapped: 'string' }
```